src/parsers/problem/DimikOJProblemParser.ts

```diff
@@ -0,0 +1,30 @@
+import { Sendable } from '../../models/Sendable';
+import { TaskBuilder } from '../../models/TaskBuilder';
+import { htmlToElement } from '../../utils/dom';
+import { Parser } from '../Parser';
+
+export class DimikOJProblemParser extends Parser {
+  public getMatchPatterns(): string[] {
+    return ['https://dimikoj.com/problems/*'];
+  }
+
+  public async parse(url: string, html: string): Promise<Sendable> {
+    const doc = htmlToElement(html);
+    const task = new TaskBuilder('DimikOJ').setUrl(url);
+
+    // don't use bangla name, it breaks code compilation in cph
+    const urlParts = url.split('/');
+    const problemNumber = urlParts.at(-2);
+    const problemName = urlParts.at(-1).replace('-', ' ').replace(/(?<=(^| ))./g, c => c.toUpperCase());
+    task.setName(`${problemNumber} ${problemName}`);
```

Comment:
In addition to CPH, I am also getting errors like this:
![image](https://github.com/jmerle/competitive-companion/assets/46617994/2fe45faf-e0d7-47f1-82b9-d0a12260943a)

Not sure how to resolve this in command line itself.

Reply:
> In addition to CPH, I am also getting errors like this:

As far as I can see there are certain characters that your compiler doesn't like to see in filenames. That may happen but is not a Competitive Companion issue. This extension only parses problem data and sends it to external tools, it is up to those tools to do something useful with it. If one of these tools creates files you cannot compile with your chosen compiler, then that's a bug in that tool and should be fixed in that tool, not in Competitive Companion.

> Also, almost all coders in Bangladesh use CPH. So, for Dimik OJ, CPH is an important extension.

No matter how popular it is, a bug in CPH should be fixed in CPH, not in Competitive Companion. There is no guarantee made by Competitive Companion that the `name` property's value is filename-friendly or compatible with all major compilers. A possible fix would be for CPH to use the [`languages.java.taskClass` property in the data sent by Competitive Companion](https://github.com/jmerle/competitive-companion#the-format) as filename (which is filename-friendly, but severely more restrictive because it must be a valid Java class too), or to do its own post-processing on the problem title to filter out incompatible characters.

Reply:
In this case I would have to revert this change.
Also, what about a `setFileName()` option that would request other extensions to set the problems codes filename to a certain filename?

Reply:
I'm not interested in extending the output format to include a filename field, I consider filenames to be something that should be dealt with at the level of the tools that use Competitive Companion's data, since those are the programs that actually create the files and because online judges don't care what you name your files locally.

Reply:
ok, should be resolved with 45a9a3117490070bbc35673b2a6ddd9abdbe0257